core/src/test/java/com/linecorp/armeria/server/Http1ServerDelayedCloseConnectionTest.java

```diff
@@ -0,0 +1,150 @@
+            assertThat(in.readLine()).isEqualTo("HTTP/1.1 200 OK");
+
+            String line;
+            boolean hasConnectionClose = false;
+            while ((line = in.readLine()) != null) {
+                if ("connection: close".equalsIgnoreCase(line)) {
+                    hasConnectionClose = true;
+                }
+                if (line.isEmpty() || line.contains(":")) {
+                    continue;
+                }
+                if (line.startsWith("OK")) {
+                    break;
+                }
+            }
+            final long readStartTimestamp = System.nanoTime();
+            final int readResult = in.read();
+            final long readDurationMillis = Duration.ofNanos(System.nanoTime() - readStartTimestamp).toMillis();
+
+            assertThat(hasConnectionClose).isTrue();
+            assertThat(readResult).isEqualTo(-1);
+
+            final long defaultHttp1ConnectionCloseDelayMillis = Flags.defaultHttp1ConnectionCloseDelayMillis();
+            assertThat(readDurationMillis).isBetween(
+                    defaultHttp1ConnectionCloseDelayMillis - 100,
+                    defaultHttp1ConnectionCloseDelayMillis + 1000
+            );
+
+            socket.close();
+            try (Socket reuseSock = new Socket()) {
+                assertThatCode(() -> reuseSock.bind(new InetSocketAddress((InetAddress) null, socketPort)))
+                        .doesNotThrowAnyException();
+            }
+        }
+    }
+
+    @Test
+    void shouldWaitForDisconnectByClientSideFirst() throws IOException {
+        try (Socket socket = new Socket("127.0.0.1", server.httpPort())) {
+            socket.setSoTimeout(100000);
+            final int socketPort = socket.getLocalPort();
+            final PrintWriter writer = new PrintWriter(socket.getOutputStream());
+            writer.print("GET /close" + " HTTP/1.1\r\n");
+            writer.print("\r\n");
+            writer.flush();
+
+            final BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            assertThat(in.readLine()).isEqualTo("HTTP/1.1 200 OK");
+
+            String line;
+            boolean hasConnectionClose = false;
+            while ((line = in.readLine()) != null) {
+                if ("connection: close".equalsIgnoreCase(line)) {
+                    hasConnectionClose = true;
+                }
+                if (line.isEmpty() || line.contains(":")) {
+                    continue;
+                }
+                if (line.startsWith("OK")) {
+                    break;
+                }
+            }
+            assertThat(hasConnectionClose).isTrue();
+            assertThat(server.server().numConnections()).isEqualTo(1);
+
+            socket.close();
+            assertThatThrownBy(
+                    () -> {
+                        final Socket reuseSock = new Socket("127.0.0.1", server.httpPort(), null, socketPort);
+                        reuseSock.close();
```

Comment:
So that we know why this logic exists

```suggestion
                        // close the socket in case initializing the socket doesn't throw an exception
                        reuseSock.close();
```

Reply:
Thanks for good suggestions 🙇